src/mongo/s/stats/joom_top.cpp

```diff
@@ -0,0 +1,96 @@
+#define MONGO_LOGV2_DEFAULT_COMPONENT ::mongo::logv2::LogComponent::kDefault
+
+#include "mongo/platform/basic.h"
+
+#include "mongo/db/curop.h"
+#include "mongo/db/jsobj.h"
+#include "mongo/db/service_context.h"
+
+#include "mongo/s/stats/joom_top.h"
+
+namespace mongo {
+
+namespace {
+
+const auto getJoomTop = ServiceContext::declareDecoration<JoomTop>();
+
+}  // namespace
+
+// static
+JoomTop& JoomTop::get(ServiceContext* service) {
+    return getJoomTop(service);
+}
+
+void JoomTop::record(OperationContext* opCtx, bool isError) {
+    auto& curOp = *CurOp::get(opCtx);
+    auto ns = curOp.getNS();
+
+    if (ns[0] == '?' || curOp.getNSS().isCommand()) {
+        return;
+    }
+
+    auto elapsedMicros = opCtx->getElapsedTime().count();
```

Comment:
From the comment of `count` method:
```
     * It is better to use durationCount<DesiredDurationType>(value), since it makes the unit of the
     * count clear at the call site.
```

Reply:
Fixed!